### PR TITLE
[DX-1358] Gateway release notes 5.4.0

### PR DESCRIPTION
**Pull request by Eopayemi**
### For internal users - Please add a Jira DX PR ticket to the subject!
[DX-1358](https://tyktech.atlassian.net/browse/DX-1358?atlOrigin=eyJpIjoiNDQ4NGZjODk2NzVmNDYxNjlkMzBlNDk0MDYzODk2MzMiLCJwIjoiamlyYS1zbGFjay1pbnQifQ)

---

### Preview Link

[preview](https://deploy-preview-4662--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/release-notes/version-5.4)

### Description
 
Release notes 5.4.0, build checks will be fixed when Tyk Charts 1.5.0 release notes merged.

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

___

### **PR Type**
Documentation

___

### **Description**
- Added comprehensive release notes for Tyk Gateway version 5.4, including sections for breaking changes, dependencies, deprecations, upgrade instructions, highlights, downloads, changelog, and security fixes.
- Updated the compatibility matrix and third-party dependencies.
- Included a new menu entry for Tyk Gateway v5.4 release notes.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>version-5.4.md</strong><dd><code>Add release notes for Tyk Gateway version 5.4</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md
<li>Added release notes for Tyk Gateway version 5.4.<br> <li> Included sections for breaking changes, dependencies, deprecations, <br>upgrade instructions, highlights, downloads, changelog, and security <br>fixes.<br> <li> Provided compatibility matrix and third-party dependencies.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4662/files#diff-3cbbef4987e7209220db6a08dd8059cab4c9df294a36572ccb1ad1b858593e2a">+244/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Update menu with Tyk Gateway v5.4 release notes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml
- Added menu entry for Tyk Gateway v5.4 release notes.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4662/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

[DX-1358]: https://tyktech.atlassian.net/browse/DX-1358?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+There are no breaking changes in this release.
+
+### Dependencies {#dependencies-5.4.0}
+<!--Required. Use this section to announce the following types of dependencies compatible with the release:
+
+Version compatibility with other components in the Tyk stack. This takes the form of a compatibility matrix and is only required for Gateway and Portal.
+
+3rd party dependencies and tools -->
+
+#### Compatibility Matrix For Tyk Components
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md:41]: @agata-wit @andyo-tyk @caroltyk @slega I have updated the recommended releases column where possible. Could you update if incorrect and update backwards compatibility column and any 3rd party dependencies for release 5.4?

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+
+Introduced a features object in API definitions for GQL APIs, including the use_immutable_headers attribute. This allows advanced header control, enabling users to add new headers, rewrite existing ones, and selectively remove specific headers. Existing APIs will have this attribute set to `false` by default, ensuring no change in behaviour. For new APIs, this attribute is true by default, facilitating smoother migration and maintaining backward compatibility.
+
+### Downloads
+- [Docker image to pull](https://hub.docker.com/r/tykio/tyk-gateway/tags?page=&page_size=&ordering=&name=v5.4.0)
+  - ```bash
+    docker pull tykio/tyk-gateway:v5.4.0
+    ``` 
+- Helm charts (update link for release 5.4)
+  - [tyk-charts v1.3]({{< ref "product-stack/tyk-charts/release-notes/version-1.3.md" >}})
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md]: @caroltyk this should be Tyk Charts 1.5, yes?

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+<li>
+<details>
+<summary>New Tyk OAS features</summary>
+
+We’ve added some more features to the Tyk OAS API, moving closer to full parity with Tyk Classic. In this release we’ve added controls that allow you: to enable or prevent generation of traffic logs at the API-level and to enable or prevent the availability of session context to middleware. We’ve also added the facility to register webhooks that will be fired in response to Gateway events. 
+</details>
+</li>
+</ul>
+
+#### Changed
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md]: @Eopayemi Are there any changes to document in the changelog? If there are no changes to. document in the release then could you remove the changed section?

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+<summary>Introduced Rate Limit Smoothing for scaling</summary>
+
+Implemented [Rate Limit Smoothing]({{< ref "getting-started/key-concepts/rate-limiting#rate-limit-smoothing" >}}) as an extension to the existing Redis Rate Limiter to gradually adjust the rate based on smoothing configuration. Two new Gateway events have been created  (`RateLimitSmoothingUp` and `RateLimitSmoothingDown`) which will be triggered as smoothing occurs. These can be used to assist with auto-scaling of upstream capacity during traffic spikes.
+</details>
+</li>
+<li>
+<details>
+<summary>Introduced ‘use_immutable_headers’ for Advanced Header Control in GraphQL APIs</summary>
+
+We've added the `use_immutable_headers` option to the GraphQL API configuration, offering advanced header transformation capabilities. When enabled, users can add new headers, rewrite existing ones, and selectively remove specific headers, allowing granular control without altering the original request. Existing APIs will default to `false`, maintaining current behaviour until ready for upgrade.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md:155]: @agata-wit have use_immutable_headers, manual schema addition and features object included in API definition been documented?

**Review by dcs3spp — CHANGES_REQUESTED** [review]
PR LGTM, with following minor suggestions:

- @agata-wit @andyo-tyk @caroltyk  @slega Could you check and update compatibility matrix, 3rd party dependencies and upgrade section in relation to release 5.4.0
- @Eopayemi Could you remove the changed section in the changelog if there no changes logged for this release
- @agata-wit I added comment related to documentation of use_immutable_headers, manual schema addition and the new features objected included in the APIDefinition.

Should Tyk Streams Beta be mentioned?

**Comment by andyo-tyk** [comment]
> Should Tyk Streams Beta be mentioned?

Tyk Streams Beta is not included in the official 5.4.0, it is still in Tyk Labs

**Review by andyo-tyk — APPROVED** [review]

**Review by carolwkc — APPROVED** [review]
compatibility matrix updated

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md (hunk 4 above)
Comment (agata-wit) [reply tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md:155]: I'm working on it.

**Review by agata-wit — APPROVED** [review]

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md (hunk 4 above)
Comment (dcs3spp) [reply tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.4.md:155]: @agata-wit when raise a PR for the feature, remember to include the links in the changelog - updated in the release notes

**Review by dcs3spp — APPROVED** [review]

**Pull request merged**